### PR TITLE
[GSoC] migrated AnkiStatsTaskHandler.createStatisticsOverview() to suspend function

### PR DESCRIPTION
**Pull request by divyansh-dxn**
_This PR is a continuation of #11556_

## Purpose / Description
AsyncTask to coroutines migration

## Approach
Replaced the usage of AsyncTask to create and display statistics overview with the suspend function.

## How Has This Been Tested?
Runs as expected on Realme 6i

https://user-images.githubusercontent.com/69595691/173676258-adf8cb58-0889-4da0-99cd-2216bf21575d.mp4

https://user-images.githubusercontent.com/69595691/173676222-c4150036-33c5-4548-89f4-56eade632110.mp4

## Learning
Use of [Mutex](https://kotlinlang.org/docs/shared-mutable-state-and-concurrency.html#mutual-exclusion) for thread safety on critical sections with suspending functions.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt [hunk 1]
```diff
@@ -51,9 +47,11 @@ import com.ichi2.libanki.stats.Stats
 import com.ichi2.libanki.stats.Stats.AxisType
 import com.ichi2.libanki.stats.Stats.ChartType
 import com.ichi2.ui.FixedTextView
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 import timber.log.Timber
-import java.util.Locale
+import java.util.*
```
Comment (ShridharGoel) [inline AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt:54]: Avoid the usage of wildcard imports

**Comment by divyansh-dxn** [comment]
@lukstbit if you could have a look, let me know if everything looks good.

**Comment by lukstbit** [comment]
@divyansh-dxn Looks good, I saw your code the day you pushed the PR. I don't have the right to approve so you'll have to wait for one of the members.

**Comment by david-allison** [comment]
@lukstbit Did you get our email about this? (Sent June 13).

If you didn't, could you get in contact [gmail: davidallisongithub], or via Discord: https://discord.gg/qjzcRTx

**Comment by lukstbit** [comment]
@david-allison I just saw it, I've answered to Arthur. I'll use the invite.

**Pull request merged**